File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine());
        
        HashSet<String> workerInCom = new HashSet<>();

        for (int i = 0; i < n; i++) {
            String[] record = br.readLine().split(" ");
            String name = record[0];
            String status = record[1];

            if (status.equals("enter")) {
                workerInCom.add(name);
            } else if (status.equals("leave")) {
                workerInCom.remove(name);
            }
        }

        ArrayList<String> result = new ArrayList<>(workerInCom);
        Collections.sort(result, Collections.reverseOrder());

        StringBuilder sb = new StringBuilder();
        for (String person : result) {
            sb.append(person).append("\n");
        }

        bw.write(sb.toString());
        bw.flush();
        br.close();
        bw.close();
    }
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

data = sys.stdin.read().splitlines()

n = int(data[0])

workerInCom = set()

for i in range(1, n + 1):
        name, status = data[i].split()
        if status == "enter":
            workerInCom.add(name)
        elif status == "leave":
            workerInCom.discard(name)

output = []

for person in sorted(workerInCom, reverse=True):
    output.append(person + "\n")

sys.stdout.write(''.join(output))